pages/coaching-dashboard.tsx

```diff
@@ -1,24 +1,56 @@
 import { useQuery } from "@apollo/client";
-import React from "react";
+import React, { useEffect } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import ProfileDetailCard from "../components/coding/studentPortal/ProfileDetailCard";
 import {
   FetchUserProfileCardResponse,
   FETCH_USER_PROFILE_CARD,
 } from "../graphql/fetchUserProfileCard";
 import Link from "next/link";
-import { userSelector, setUserList } from "../redux/userSlice";
+import { userSelector, setUserList, setEarnedBadges } from "../redux/userSlice";
+import {
+  FetchEarnedBadges,
+  FETCH_EARNED_BADGES,
+} from "../graphql/fetchEarnedBadges";
+import { transformBadgesEarned } from "./api/coachingDashboard";
 const coachingDashboard = () => {
   const dispatch = useDispatch();
 
-  const { userList } = useSelector(userSelector);
+  const { userList, earnedBadges } = useSelector(userSelector);
 
   const { loading, data } = useQuery<FetchUserProfileCardResponse>(
     FETCH_USER_PROFILE_CARD,
     {
-      onCompleted: () => dispatch(setUserList(data.users)),
+      onCompleted: () =>
+        data.users.length > 0 ? dispatch(setUserList(data.users)) : loading,
     }
   );
+
+  const enrolledUsers = userList.map((user) => user.id);
+
+  const {} = useQuery<FetchEarnedBadges>(FETCH_EARNED_BADGES, {
+    variables: {
+      enrolledIds: enrolledUsers,
+    },
+    onCompleted: (data) => {
+      dispatch(setEarnedBadges(data.user_coding_badges));
+    },
+  });
+
+  useEffect(() => {
+    const aggregatedBadgeCount = earnedBadges.reduce((acc, badgeId) => {
```

Comment:
This is cool that you figured out how to transform the data, but it's confusing for new developers. I think we can figure out an easier way.

Reply:
great idea.  I removed it.  thank you!